spec/helpers/regions_helper_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+RSpec.describe Reports::RegionsHelper, type: :helper do
+  describe "#recent_bp_log" do
+    it "orders bps by date descending" do
+      Timecop.freeze("1 Jul 2021 1PM UTC") do
+        bp_1 = create(:blood_pressure, recorded_at: 2.day.ago)
+        bp_2 = create(:blood_pressure, recorded_at: 1.day.ago)
+
+        expect(recent_bp_log(BloodPressure.all)).to eq([bp_2, bp_1])
+      end
+    end
+
+    it "orders bps by time of day ascending for BPs on the same date" do
+      Timecop.freeze("1 Jul 2021 1PM UTC") do
+        bp_1 = create(:blood_pressure, recorded_at: 20.minutes.ago)
+        bp_2 = create(:blood_pressure, recorded_at: 10.minutes.ago)
+
+        expect(recent_bp_log(BloodPressure.all)).to eq([bp_1, bp_2])
+      end
+    end
+
+    context "respects the reporting timezone for ordering" do
+      it "BPs in different days in reporting time zone but same day in UTC are ordered descending" do
+        bp_1 = create(:blood_pressure, recorded_at: Time.parse("1 July 2021 11:30PM IST"))
+        bp_2 = create(:blood_pressure, recorded_at: Time.parse("2 July 2021 12:30AM IST"))
```

Comment:
I think you probably want to use `Time.zone.parse` here, otherwise I'm not sure the times you create will actually respect the time zones you are providing.

Reply:
![Screen Shot 2021-07-22 at 12 39 13 PM](https://user-images.githubusercontent.com/4241399/126601650-a6564725-0d39-4420-adf0-77a2d4d7eb4d.png)

Looks like both work, but I'd be more comfortable dealing with the ActiveSupport times anyway.

Reply:
@harimohanraj89 That probably worked because your local timezone is IST. It doesn't work on a UTC instance.

Reply:
That's why I used EST timestamps as well 😅  But I agree, let's use `Time.zone`